util/patch/patch.go

```diff
@@ -86,49 +83,166 @@ func (h *Helper) Patch(ctx context.Context, resource runtime.Object) error {
 		return errors.Errorf("expected non-nil resource")
 	}
 
-	// If the object is already unstructured, we need to perform a deepcopy first
-	// because the `DefaultUnstructuredConverter.ToUnstructured` function returns
-	// the underlying unstructured object map without making a copy.
-	if _, ok := resource.(runtime.Unstructured); ok {
-		resource = resource.DeepCopyObject()
-	}
-
 	// Convert the resource to unstructured to compare against our before copy.
-	after, err := runtime.DefaultUnstructuredConverter.ToUnstructured(resource)
+	raw, err := runtime.DefaultUnstructuredConverter.ToUnstructured(resource.DeepCopyObject())
 	if err != nil {
 		return err
 	}
+	h.after = &unstructured.Unstructured{Object: raw}
+
+	// Determine if the object has status.
+	if unstructuredHasStatus(h.after) {
+		if h.opts.IncludeStatusObservedGeneration {
+			// Set status.observedGeneration if we're asked to do so.
+			if err := unstructured.SetNestedField(h.after.Object, h.after.GetGeneration(), "status", "observedGeneration"); err != nil {
+				return err
+			}
+
+			// Restore the changes back to the original object.
+			if err := runtime.DefaultUnstructuredConverter.FromUnstructured(h.after.Object, resource); err != nil {
+				return err
+			}
+		}
+	}
+
+	return kerrors.NewAggregate([]error{
+		h.patch(ctx, resource),
+		h.patchStatus(ctx, resource),
+		h.patchStatusConditions(ctx, resource),
+	})
+}
+
+// patch issues a patch for everything in the object except for the status subresource.
```

Comment:
Does this patch subresources other than status? Or more, does the client.Patch do so, I had thought it didn't patch any subresources

Reply:
No, but this includes metadata as well

Reply:
Fixed the text a little, ptal

Reply:
Lgtm